crates/project-model/src/workspace.rs

```diff
@@ -270,7 +268,9 @@ impl ProjectWorkspace {
 
         tracing::info!(workspace = %cargo_toml, src_root = ?sysroot.rust_lib_src_root(), root = ?sysroot.root(), "Using sysroot");
         progress("querying project metadata".to_owned());
-        let toolchain_config = QueryConfig::Cargo(&sysroot, cargo_toml);
+        let config_file = cargo_config_file::read(cargo_toml, extra_env, &sysroot);
+        let config_file_ = config_file.clone();
+        let toolchain_config = QueryConfig::Cargo(&sysroot, cargo_toml, &config_file_);
```

Comment:
```suggestion
        let toolchain_config = QueryConfig::Cargo(&sysroot, cargo_toml, &config_file.clone());
```
does this not work? I'd assume this constructs a temporary

Reply:
Unfortunately, since this is not a function call but a tuple ctor, that doesn't work 😅 
```rust
QueryConfig::Cargo {
    0: &sysroot,
    1: cargo_toml,
    2: &config_file.clone(),
}
```
might work but I think it's not desirable